### ISSUE TITLE
renaming a torrent's subfolder can cause problems with torrent completion

### ISSUE THREAD

**Issue by as-muncher** [BiglySoftware/BiglyBT#3166]
Java 1.8.0_202 (64 bit)
  Oracle Corporation
c:\program files\biglybt\jre

SWT v4942r22, win32, zoom=100, dpi=120
Windows 10 v10.0, amd64 (64 bit)
B3.5.0.0/4 az3

I had/have torrents, where they will save to 99.8% completion, but not fully, even though all the data is there.  I'd see that in that one piece that can't complete, is where the screenshots are.  I knew all the data was there.  I saw a number of .jpg.!qB files that BiglyBT had created, and I also saw the fully complete .jpg files.  And then one video file had a .mp4.!qB extension.  So what I did is I deleted all the files with the .!qB extension, and checked again.  Now, the torrent shows as 88.0% complete.  The problem has to do with the save name of the torrent root directory.  This is the name that I choose you open a new torrent and it wants to know the save location etc.  I choose my regular save location, but change the subfolder name to something else.  I see this problem even when BiglyBT saves to Example-2.  The way qbittorrent does it, is it will save to the directory you choose, but if you choose a subfolder that is different, qbittorrent will create the torrent's root folder and then the subfolders, so \DifferentName\Example\Screens and \DifferentName\Example\Videos.  But BiglyBT, if you choose a different subfolder name, will save to \DifferentName\Screens and \DifferentName\Videos.  I think the torrent can't complete because that Example directory has to be there.  If I move the Screens and Videos folders and files all completed into that \DifferentName subdirectory BiglyBT shows data as 88% complete.  However, if I don't rename the saving subdirectory, that torrent root folder, but keep it saved as Example, whatever the default save name is, then it shows as 100% complete, seeds.

So, in short, changing the torrent save name causes the torrent to not complete, even though all the data is there, because that directory needs to be found.

**Comment by parg** [comment]
Ànother one of your rambling reports with no definite explanation of the problem. I can't work on this. Seriously, just stick to qbittorrent, it sounds awesome.

**Comment by as-muncher** [comment]
What would help you make this more helpful?  I'm sorry I ramble.

**Comment by parg** [comment]
First I don't care what qbittorrent does, 

I develop and support BiglyBT. I am pretty much the only person doing this. It is a hobby, I'm not paid to do it.

I don't have time to try and figure out people's problems, I need a clear set of instructions to reproduce the problem reliably. For example:

Step 1: add this torrent
Step 2: these are all of my configuration settings relevant to the problem
Step 3: do this
Step 4: do that
Result: The problem/bug occurs.

If I can reproduce a bug then I can fix it (or at least decide if it is not going to be fixed)

If I can't reproduce it simply and reliably then I can't fix it.

I can't download random torrents full of who-knows-what content, legally that is not possible. So ideally the problem can be reproduced with a public domain torrent (if one is needed)

If you can't reproduce the problem reliably then please don't create an issue.

Also a lot of your problems seem to be to do with importing partially completed downloads from qbittorrent. That's generally not something I support as I have no idea what's going on. You need to create the issue within BiglyBT from a clean state.

Also you know that there are limits within BiglyBT regarding the maximum number of "file links" per torrent. This is not something I am going to fix anytime soon so if it something to do with that then please don't raise an issue.

**Comment by as-muncher** [comment]
Thanks for your response.
I merely use the file extension for incomplete files as ".!qB" so perhaps it looks like I'm importing a lot from qbittorrent.  Of course, I need to sometimes in order for swarm merging to happen.  But woe to me if BiglyBT and qbittorrent try to access the files at the same time - data loss will result.  I am simply making you aware of issues with your software. It's not meant to put you down or anything like that.  But perhaps I can be more rigorous and give you more to work with, if possible.  But on my end, I produce an Issue because I'm letting you know of something that needs to be fixed.  It's up to you to choose whether to do it or not.  If you choose to close the Issue as "not planned", totally up to you.  But perhaps there may be others who notice the same problem.  I think the reason there seem to be so many problems, is because possibly you are testing your software without an extension for incomplete files, and so I use that all the time.  It is so annoying when people don't download something completely and then seed it as complete when it's not.  So perhaps when you fix something, maybe you are fixing files without some sort of incomplete-file extension.  I don't know.
